src/kudu/consensus/raft_consensus.cc

```diff
@@ -3443,12 +3453,16 @@ void RaftConsensus::CompleteConfigChangeRoundUnlocked(ConsensusRound* round, con
 
 void RaftConsensus::EnableFailureDetector(boost::optional<MonoDelta> delta) {
   if (PREDICT_TRUE(FLAGS_enable_leader_failure_detection)) {
+    persistent_vars_->enable_failure_detector();
+    CHECK_OK(persistent_vars_->Flush());
     failure_detector_->Start(std::move(delta));
   }
 }
 
 void RaftConsensus::DisableFailureDetector() {
   if (PREDICT_TRUE(FLAGS_enable_leader_failure_detection)) {
+    persistent_vars_->disable_failure_detector();
```

Comment:
This doesn't look right. We call DisableFailureDetector() when we become leader. I think the right usage is sort of inverted here. Enabling should be a no-op if the persistent var is set to false, and we need an API so that when OLM needs to disable the FD then it would set the flag and call consensus_->DisableFailureDetector()